### ISSUE TITLE
ncurses terminfo has some issues

### ISSUE THREAD

**Issue by jclulow** [TritonDataCenter/pkgsrc#248]
There are a couple of issues with the `terminfo.src` in the current ncurses source.

First, the unfortunate reality is that lots of people use `xterm` and `xterm-256color` for terminals other than a strictly compliant `xterm`.  The current database appears to include a `rep` capability, but many terminals don't implement this correctly.  The graphical corruption is pretty obvious if you use something like `ncdu` which will make use of the repeating functionality, in a terminal like `urxvt`.

Ditching the `rep` capability from the `xterm` definition causes ncurses to fall back to something slightly less efficient, but widely supported: drawing the character multiple times.  This seems a low risk, high value patch we could carry.

Second, `tmux` encourages users to use `TERM=screen` or `screen-256color`.  A change was made just prior to the 6.2 release, it seems, that causes the old `screen` definition to be ditched in favour of one that inherits the `ecma+index` base:

```
# 2019-06-30
...
#       + use ecma+index in screen, st -TD?
```

This definition includes a `rin` capability that flat out doesn't work with `tmux`:

```
# ECMA-48 does not include the VT100 indexing and scroll-margins.  It has its
# own variation.
ecma+index|ECMA-48 scroll up/down,
        indn=\E[%p1%dS, rin=\E[%p1%dT,
```

(i.e., `CSI n T` does nothing)

In a subsequent version, this was apparently corrected:

```
20190803
...
        + amend the change to screen, because tmux relies upon that entry
          and does not support that feature (Debian #933572) -TD
```

Unfortunately I think we only get a new `terminfo.src` when we get a new ncurses release.  I think it'd be a good idea for `pkgsrc` to contain a wholesale copy of `terminfo.src` which can be somewhat detached form the ncurses update process.  We could also carry local modifications on the file.  This is, I believe, what most distributions end up having to do, to manage the churn which is at times unhelpful.

What do you think?

**Comment by jperkin** [comment]
Both issues fixed in https://github.com/NetBSD/pkgsrc/commit/4a217298a4bdba7c47d1311a51540f6e6bad42d5 and https://github.com/NetBSD/pkgsrc/commit/e8a4d5a700047e40346db20c2e30be5a9a222788.

For now I'm going to hold off on maintaining our own copy of `terminfo.src` as it's not without risk (it's likely we'd maintain it as a patch against upstream pkgsrc which would be inherently vulnerable to bit-rot), but if we see regressions again then we'll strongly consider it.

A fixed `ncurses-6.2nb2` package will be available in the next trunk build, as well as our 2020Q1 release.

**Comment by jclulow** [comment]
Thanks!
